File: NSNotifsExample/Objective-C/FirstVC.m
#import "FirstVC.h"


@implementation FirstVC


- (void)viewDidLoad {

	[super viewDidLoad];

	self.magicButton = [UIButton new];
	self.magicButton.titleLabel.font = [UIFont systemFontOfSize:15];
	self.magicButton.backgroundColor = UIColor.systemIndigoColor;
	self.magicButton.layer.cornerCurve = kCACornerCurveContinuous;
	self.magicButton.layer.cornerRadius = 20;
	self.magicButton.translatesAutoresizingMaskIntoConstraints = NO;
	[self.magicButton setTitle : @"Tap me and switch tabs to see magic" forState:UIControlStateNormal];
	[self.magicButton addTarget : self action:@selector(notificationSender:) forControlEvents:UIControlEventTouchUpInside];
	[self.view addSubview:self.magicButton];

	[self.magicButton.centerXAnchor constraintEqualToAnchor : self.view.centerXAnchor].active = YES;
	[self.magicButton.centerYAnchor constraintEqualToAnchor : self.view.centerYAnchor].active = YES;
	[self.magicButton.widthAnchor constraintEqualToConstant : 260].active = YES;
	[self.magicButton.heightAnchor constraintEqualToConstant : 40].active = YES;

}


- (void)notificationSender:(id)sender {

	[NSNotificationCenter.defaultCenter postNotificationName:@"fireNotificationDone" object:nil];

}


@end

File: NSNotifsExample/Swift/FirstVC.swift
import UIKit


class FirstVC: UIViewController {

	/*--- instantiate UIButton so we can access all 
	of it's properties and create our button ---*/

	var magicButton = UIButton()

	override func viewDidLoad() {

		super.viewDidLoad() // call the original implementation of viewDidLoad()

		// create our button

		magicButton = UIButton()
		magicButton.setTitle("Tap me and switch tabs to see magic", for: .normal)
		magicButton.titleLabel!.font = UIFont.systemFont(ofSize: 15)
		magicButton.backgroundColor = .systemIndigo
		magicButton.layer.cornerCurve = .continuous
		magicButton.layer.cornerRadius = 20
		magicButton.translatesAutoresizingMaskIntoConstraints = false
		magicButton.addTarget(self, action:#selector(notificationSender), for: .touchUpInside)
		view.addSubview(magicButton)

		// call our setupLayout method, yes it's called method here because it's within the scope of our class

		setupLayout()

	}

	private func setupLayout() {

		// proper UI layout

		magicButton.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
		magicButton.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
		magicButton.widthAnchor.constraint(equalToConstant: 280).isActive = true
		magicButton.heightAnchor.constraint(equalToConstant: 40).isActive = true

	}

	/*--- we use @objc because "selector" is an Objective-C concept only,
	so we need to make it visible to the Objective-C runtime ---*/

	@objc private func notificationSender() {

		// fire the notification when clicking the button

		NotificationCenter.default.post(name: Notification.Name("fireNotificationDone"), object: nil)
		
	}

}
